agent/background.py

```
@@ -155,9 +184,122 @@ def _handle(self, task: BackgroundTask) -> None:
+        # Heartbeat updater — called by the agent job to report progress
+        def heartbeat(phase: str = "", message: str = "") -> None:
+            task.last_heartbeat_at = _now()
+            if message:
+                task.progress_message = message
+                log.debug("BG task %s heartbeat: %s — %s", task.task_id, phase, message)
+            # Refresh notification with progress if callback wired
+            if self._on_task_complete:
+                try:
+                    self._on_task_complete(task)
+                except Exception:
+                    pass
```

Ruff (0.15.15)

[error] `try`-`except`-`pass` detected, consider logging the exception

(S110)